cmake/compiler/clang/target_arm64.cmake

```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Just as a note, this file is a copy of https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/gcc/target_arm64.cmake except with the `-mabi` flag bits removed. As best I can tell, clang and gcc expect slightly different things for `-mabi` for aarch64. Quoting from https://reviews.llvm.org/D104931: "The situation around -mabi for AArch64 is a bit awkward: Clang uses it to select the ARM64 calling convention (`-mabi=aapcs` vs `-mabi=darwinpcs`) while GCC uses it to select between LP64 and ILP32."

IIUC, clang determines lp64 vs ilp32 based on the triple passed, so aarch64-none-elf should end up "selecting" lp64. Leaving the `-mabi=lp64` bits in results in errors for clang.